tests/integration/bank/keeper/keeper_test.go

```diff
@@ -0,0 +1,1667 @@
+	maccPerms[randomPerm] = []string{"random"}
+	authKeeper := authkeeper.NewAccountKeeper(
+		appCodec, app.GetKey(types.StoreKey), authtypes.ProtoBaseAccount,
+		maccPerms, sdk.Bech32MainPrefix, authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+	)
+	keeper := keeper.NewBaseKeeper(
+		appCodec, app.GetKey(types.StoreKey), authKeeper, blockedAddrs, authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+	)
+
+	return authKeeper, keeper
+}
+
+func (suite *IntegrationTestSuite) SetupTest() {
+	suite.app = simapp.Setup(suite.T(), false)
+	suite.ctx = suite.app.BaseApp.NewContext(false, tmproto.Header{Time: time.Now()})
+
+	suite.Require().NoError(suite.app.AccountKeeper.SetParams(suite.ctx, authtypes.DefaultParams()))
+	suite.Require().NoError(suite.app.BankKeeper.SetParams(suite.ctx, types.DefaultParams()))
+
+	queryHelper := baseapp.NewQueryServerTestHelper(suite.ctx, suite.app.InterfaceRegistry())
+	types.RegisterQueryServer(queryHelper, suite.app.BankKeeper)
+	queryClient := types.NewQueryClient(queryHelper)
+	types.RegisterInterfaces(suite.app.InterfaceRegistry())
+
+	suite.queryClient = queryClient
+	suite.msgServer = keeper.NewMsgServerImpl(suite.app.BankKeeper)
+}
+
+func (suite *IntegrationTestSuite) TestSupply() {
+	ctx := suite.ctx
+
+	require := suite.Require()
+
+	// add module accounts to supply keeper
+	authKeeper, keeper := suite.initKeepersWithmAccPerms(make(map[string]bool))
+
+	genesisSupply, _, err := keeper.GetPaginatedTotalSupply(ctx, &query.PageRequest{})
+	require.NoError(err)
+
+	initialPower := int64(100)
+	initTokens := suite.app.StakingKeeper.TokensFromConsensusPower(ctx, initialPower)
+	initCoins := sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, initTokens))
+
+	// set burnerAcc balance
+	authKeeper.SetModuleAccount(ctx, burnerAcc)
+	require.NoError(keeper.MintCoins(ctx, authtypes.Minter, initCoins))
+	require.NoError(keeper.SendCoinsFromModuleToAccount(ctx, authtypes.Minter, burnerAcc.GetAddress(), initCoins))
+
+	total, _, err := keeper.GetPaginatedTotalSupply(ctx, &query.PageRequest{})
+	require.NoError(err)
+
+	expTotalSupply := initCoins.Add(genesisSupply...)
+	require.Equal(expTotalSupply, total)
+
+	// burning all supplied tokens
+	err = keeper.BurnCoins(ctx, authtypes.Burner, initCoins)
+	require.NoError(err)
+
+	total, _, err = keeper.GetPaginatedTotalSupply(ctx, &query.PageRequest{})
+	require.NoError(err)
+	require.Equal(total, genesisSupply)
+}
+
+func (suite *IntegrationTestSuite) TestSendCoinsFromModuleToAccount_Blocklist() {
+	ctx := suite.ctx
+
+	// add module accounts to supply keeper
+	addr1 := sdk.AccAddress([]byte("addr1_______________"))
+	_, keeper := suite.initKeepersWithmAccPerms(map[string]bool{addr1.String(): true})
+
+	suite.Require().NoError(keeper.MintCoins(ctx, minttypes.ModuleName, initCoins))
+	suite.Require().Error(keeper.SendCoinsFromModuleToAccount(
+		ctx, minttypes.ModuleName, addr1, initCoins,
+	))
+}
+
+func (suite *IntegrationTestSuite) TestSupply_SendCoins() {
```

Comment:
We have the exact same tests, once here with real dependencies (== integration tests), and once in `x/bank/keeper` with mocks (== unit tests).

I'm not sure I really like the duplication, especially if we do this for all modules.

If we keep these integration tests here, I actually see no added value in the `x/bank/keeper` mocked unit tests. Maybe we can delete those mocked unit tests, or rewrite them from scratch in a better way (e.g. BDD discussed in Slack).

Reply:
I like the idea because there is some duplication, but both styles of testing have some value.  Should we address rewriting bank unit tests in future work or do you consider that blocking for merging these (deleted) integration tests back?

Reply:
I'll add your points to the ADR